cmd/kubelet/app/server.go

```diff
@@ -677,15 +677,17 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies, featureGate f
 		}
 
 		if reservedSystemCPUs.Size() > 0 {
-			// at cmd option valication phase it is tested either --system-reserved-cgroup or --kube-reserved-cgroup is specified, so overwrite should be ok
+			// In case --kube-reserved-cgroup or --system-reserved-cgroup is specified, it depends on the --enforce-node-allocatable whether --reserved-cpus
+			// should enforce the cpuset of those cgroups. Here we set just the cpu shares.
 			klog.Infof("Option --reserved-cpus is specified, it will overwrite the cpu setting in KubeReserved=\"%v\", SystemReserved=\"%v\".", s.KubeReserved, s.SystemReserved)
-			if s.KubeReserved != nil {
-				delete(s.KubeReserved, "cpu")
-			}
 			if s.SystemReserved == nil {
 				s.SystemReserved = make(map[string]string)
 			}
+			if s.KubeReserved == nil {
+				s.KubeReserved = make(map[string]string)
+			}
 			s.SystemReserved["cpu"] = strconv.Itoa(reservedSystemCPUs.Size())
+			s.KubeReserved["cpu"] = strconv.Itoa(reservedSystemCPUs.Size())
```

Comment:
I think, ideally, you would have these cpusets be discrete, no?

Reply:
That would definitely provide more options to the system administrator -- the kubernetes components and the system components could be separated on the system level. However, at the moment we only have the `--reserved-cpus` flag to indicate cpuset not available to pods.

Reply:
So why not have `--system-reserved-cpus` and `--kube-reserved-cpus`?